src/outputs.rs

```diff
@@ -0,0 +1,172 @@
+use std::process;
+
+use crate::block_header::{BlockAuxilaryOutput, VerifierParams};
+use crate::requests::{BatchL1Data, L1BatchAndProofData};
+use circuit_definitions::snark_wrapper::franklin_crypto::bellman::pairing::bn256::Fr;
+
+use serde::ser::SerializeStruct;
+use serde::Serialize;
+use serde_repr::Serialize_repr;
+
+#[derive(Serialize_repr, PartialEq, Clone, Debug)]
+#[repr(u64)]
+pub enum StatusCode {
+    Success = 0,
+    InvalidNetwork,
+    NoRPCProvided,
+    FailedToDeconstruct,
+    FailedToGetDataFromL1,
+    FailedToFindCommitTxn,
+    InvalidLog,
+    FailedToGetTransactionReceipt,
+    FailedToGetBatchCommitment,
+    ProofDoesntExist,
+    FailedToFindProveTxn,
+    InvalidTupleTypes,
+    FailedToCallRPC,
+    VerificationKeyHashMismatch,
+    FailedToDownloadVerificationKey,
+    FailedToWriteVerificationKeyToDisk,
+    ProofVerificationFailed,
+}
+
+#[derive(Default)]
+pub struct VerificationKeyHashJsonOutput {
+    layer_1_vk_hash: [u8; 32],
+    computed_vk_hash: [u8; 32],
+}
+
+pub fn construct_vk_output(
+    layer_1_vk_hash: [u8; 32],
+    computed_vk_hash: [u8; 32],
+) -> VerificationKeyHashJsonOutput {
+    VerificationKeyHashJsonOutput {
+        layer_1_vk_hash,
+        computed_vk_hash,
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename = "data", rename_all = "camelCase")]
+pub struct DataJsonOutput {
+    pub batch_l1_data: BatchL1Data,
+    pub aux_input: BlockAuxilaryOutput,
+    pub verifier_params: VerifierParams,
+    pub verification_key_hash: VerificationKeyHashJsonOutput,
+    pub public_input: Fr,
+    pub is_proof_valid: bool,
+}
+
+impl From<L1BatchAndProofData> for DataJsonOutput {
+    fn from(batch: L1BatchAndProofData) -> Self {
+        Self {
+            batch_l1_data: batch.batch_l1_data,
+            aux_input: batch.aux_output,
+            verifier_params: batch.verifier_params,
+            verification_key_hash: VerificationKeyHashJsonOutput::default(),
+            public_input: Fr::default(),
+            is_proof_valid: false,
+        }
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BoojumCliJsonOutput {
+    pub status_code: StatusCode,
+    pub batch_number: u64,
+    pub data: Option<DataJsonOutput>,
+}
+
+impl Serialize for BatchL1Data {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut state = serializer.serialize_struct("BatchL1Data", 6)?;
+        let _ = state.serialize_field("prevStateRoot", &hex::encode(self.previous_root.as_slice()));
```

Comment:
why do you need "let _ "?

Reply:
we know that the values will be present, it seemed cleaner than using a bunch of expects to handle the compiler complaining about an unused result